akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/AccountExample2.java

```diff
@@ -0,0 +1,208 @@
+    public final double amount;
+
+    Deposited(double amount) {
+      this.amount = amount;
+    }
+
+    @Override
+    public Account applyTo(Account state) {
+      return state.deposited(this);
+    }
+  }
+  public static class Withdrawn implements AccountEvent {
+    public final double amount;
+
+    Withdrawn(double amount) {
+      this.amount = amount;
+    }
+
+    @Override
+    public Account applyTo(Account state) {
+      return state.withdrawn(this);
+    }
+  }
+  public static class AccountClosed implements AccountEvent {
+    @Override
+    public Account applyTo(Account state) {
+      return state.closed(this);
+    }
+  }
+
+  interface Account {
+    default Account created(AccountCreated event) {
+      throw new IllegalStateException();
+    }
+
+    default Account deposited(Deposited event) {
+      throw new IllegalStateException();
+    }
+
+    default Account withdrawn(Withdrawn event) {
+      throw new IllegalStateException();
+    }
+
+    default Account closed(AccountClosed event) {
+      throw new IllegalStateException();
+    }
+  }
+
+  public enum EmptyAccount implements Account {
+    INSTANCE;
+
+    @Override
+    public Account created(AccountCreated event) {
+      return new OpenedAccount(0.0);
+    }
+  }
+
+  public static class OpenedAccount implements Account {
+    public final double balance;
+
+    OpenedAccount(double balance) {
+      this.balance = balance;
+    }
+
+    @Override
+    public Account deposited(Deposited event) {
+      return new OpenedAccount(balance + event.amount);
+    }
+
+    @Override
+    public Account withdrawn(Withdrawn event) {
+      return new OpenedAccount(balance - event.amount);
+    }
+
+    @Override
+    public Account closed(AccountClosed event) {
+      return ClosedAccount.INSTANCE;
+    }
+
+    private boolean canWithdraw(double amount) {
+      return (balance - amount) >= 0.0;
+    }
+
+    private boolean isCloseable() {
+      return balance == 0.0;
+    }
+  }
+
+  public enum ClosedAccount implements Account {
+    INSTANCE
+  }
+
+  public static Behavior<AccountCommand> behavior(String accountNumber) {
+    return Behaviors.setup(context -> new AccountExample2(context, accountNumber));
+  }
+
+  public AccountExample2(ActorContext<AccountCommand> context, String accountNumber) {
+    super(new PersistenceId(accountNumber));
+  }
+
+  @Override
+  public Account emptyState() {
+    return EmptyAccount.INSTANCE;
+  }
+
+  private CommandHandlerBuilder<AccountCommand, AccountEvent, EmptyAccount, Account> initialHandler() {
+    return commandHandlerBuilder(EmptyAccount.class)
+      .matchCommand(CreateAccount.class, this::create);
+  }
+
+  private CommandHandlerBuilder<AccountCommand, AccountEvent, OpenedAccount, Account> openedAccountHandler() {
+    return commandHandlerBuilder(OpenedAccount.class)
+        .matchCommand(Deposit.class, this::deposit)
+        .matchCommand(Withdraw.class, this::withdraw)
```

Comment:
I like the "method handle" syntax and will start using it more. To be honest I thought lambda syntax was more well known and this was more like foreign syntax sugar.

Reply:
I'm not too sure about what's more familiar, but I think for someone unfamiliar with both, this is a little easier to infer what it does. Effective Java 3rd Ed. recommends to "Prefer method references to lambdas".